### PR TITLE
Reworked TypeScript generation

### PR DESCRIPTION
**Pull request by cblech**
Added multi scene support
Added scene in scene support
Added GLTFShape from asset manager support

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-861m4239x Integrate assets into Typescript generation](https://app.clickup.com/t/861m4239x)

File: Assets/Scripts/EditorState/AvailableComponentsState.cs [hunk 1]
```diff
@@ -31,41 +39,66 @@ public struct AvailableComponent
 
         private readonly IReadOnlyList<AvailableComponent> buildInComponents = new List<AvailableComponent>
         {
+            new AvailableComponent
+            {
+                category = "Build in",
```
Comment (tcdrop) [inline Assets/Scripts/EditorState/AvailableComponentsState.cs]: I think the correct spelling is "Builtin"

File: Assets/Scripts/System/BuilderAssetLoaderSystem.cs [hunk 2]
```diff
@@ -94,16 +94,18 @@ private struct AssetPacksMetrics
         private EditorEvents _editorEvents;
         private LoadGltfFromFileSystem _loadGltfFromFileSystem;
         private IWebRequestSystem _webRequestSystem;
+        private IPathState _pathState;
```
Comment (tcdrop) [inline Assets/Scripts/System/BuilderAssetLoaderSystem.cs]: Naming convention without _

File: Assets/Scripts/System/BuilderAssetLoaderSystem.cs [hunk 3]
```diff
@@ -328,6 +331,33 @@ public AssetData GetDataById(Guid id)
...
+                return null;
+            }
+
+            // Copy all files into the build path
+            foreach (var (path, hash) in data.contentsPathToHash)
+            {
+                var filePath = await assetDownLoader.GetFileFromHash(hash);
+
+                var destFileName = Path.Combine(modelBuildPath, path);
+                Directory.CreateDirectory(Path.GetDirectoryName(destFileName) ?? throw new InvalidOperationException());
```
Comment (tcdrop) [inline Assets/Scripts/System/BuilderAssetLoaderSystem.cs]: Place file system operations in try-catch-statement

**Review by tcdrop — CHANGES_REQUESTED** [review]

File: Assets/Scripts/EditorState/AvailableComponentsState.cs (hunk 1 above)
Comment (cblech) [reply Assets/Scripts/EditorState/AvailableComponentsState.cs]: Yes, but with a space or hyphen.
"Built in" or "Built-in"

**Pull request merged**